packages/sdk/src/evm/core/classes/marketplacev3-direct-listings.ts

```diff
@@ -0,0 +1,833 @@
+import {
+  cleanCurrencyAddress,
+  fetchCurrencyValue,
+  normalizePriceValue,
+  setErc20Allowance,
+} from "../../common/currency";
+import {
+  getAllInBatches,
+  handleTokenApproval,
+  isTokenApprovedForTransfer,
+} from "../../common/marketplacev3";
+import { fetchTokenMetadataForContract } from "../../common/nft";
+import {
+  InterfaceId_IERC1155,
+  InterfaceId_IERC721,
+} from "../../constants/contract";
+import {
+  DirectListingInputParams,
+  DirectListingInputParamsSchema,
+} from "../../schema/marketplacev3/direct-listings";
+import { DirectListingV3, MarketplaceFilter } from "../../types";
+import { TransactionResult, TransactionResultWithId } from "../types";
+import { ContractWrapper } from "./contract-wrapper";
+import type {
+  IERC1155,
+  IERC165,
+  IERC721,
+  IDirectListings,
+  DirectListingsLogic,
+} from "@thirdweb-dev/contracts-js";
+import ERC165Abi from "@thirdweb-dev/contracts-js/dist/abis/IERC165.json";
+import ERC721Abi from "@thirdweb-dev/contracts-js/dist/abis/IERC721.json";
+import ERC1155Abi from "@thirdweb-dev/contracts-js/dist/abis/IERC1155.json";
+import {
+  NewListingEvent,
+  UpdatedListingEvent,
+} from "@thirdweb-dev/contracts-js/dist/declarations/src/DirectListingsLogic";
+import { ThirdwebStorage } from "@thirdweb-dev/storage";
+import { BigNumber, BigNumberish, Contract } from "ethers";
+import invariant from "tiny-invariant";
+
+/**
+ * Handles direct listings
+ * @public
+ */
+export class MarketplaceV3DirectListings {
+  private contractWrapper: ContractWrapper<DirectListingsLogic>;
+  private storage: ThirdwebStorage;
```

Comment:
can you add ContractEvents / ContractEncoder and ContractInterceptor to the plugins as well?